spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/BatchDataReader.java

```diff
@@ -126,7 +138,8 @@ CloseableIterator<ColumnarBatch> open(FileScanTask task) {
   }
 
   private SparkDeleteFilter deleteFilter(FileScanTask task) {
-    return task.deletes().isEmpty() ? null : new SparkDeleteFilter(task, table().schema(), expectedSchema);
+    LOG.debug("is task.deletes() empty? {}", task.deletes().isEmpty());
```

Comment:
I would probably just break up the ternary operator below and log when creating the Delete filter?

Reply:
Done.